File: 이건희/level2/카펫.py
def solution(brown, yellow):
    result = []
    a, b = 0, 0
    
    for i in range(1, yellow + 1):
        if yellow % i == 0:
            a = i
            b = yellow / i
            if 2 * (a + b) + 4 == brown:
                result.append(a + 2)
                result.append(b + 2)
                break
                
    result.sort(reverse = True)
    return result

File: 김찬우/프로그래머스 /Lv.2/카펫.java
import java.lang.Math;
class Solution {
    public int[] solution(int brown, int yellow) {
        
        // 아이디어 
        // yellow를 한줄 두줄로 해서 전부탐색 
        
        int[] answer = new int[2];
        
        
        for(int i = 1 ; i <= yellow; i++){
            
            if(yellow % i !=0){
                continue;
            }
        
            int x = yellow / i;
            int y = i;
            
            if( (x+2)*2 + y*2 ==  brown){
                
                answer[0] = Math.max(x+2,y+2);
                answer[1] = Math.min(x+2,y+2);
                
            }
            
        }
        
        
        

        return answer;
    }
}

      
